pyomo/repn/plugins/gams_writer.py

```diff
@@ -201,46 +312,62 @@ def _print_model( self, model, output_file, solver_capability, var_list,
             var = var_list[i]
             i += 1
             v = symbolMap.getObject(var)
-            try:
+            if not v.is_expression():
                 if v.is_binary():
                     binary.append(var)
                 elif v.is_integer():
                     if v.bounds == (0,1):
                         binary.append(var)
+                    elif v.lb is not None and v.lb >= 0:
+                        posInts.append(var)
                     else:
-                        ints.append(var)
+                        otherInts.append(var)
                 elif v.lb == 0:
                     positive.append(var)
                 else:
                     reals.append(var)
-            except AttributeError:
-                # This is an Expression object
-                body = StringIO()
-                v.to_string(body)
+            else:
+                # Same domain violation check as above
+                try:
+                    value(v.expr)
+                except:
+                    raise RuntimeError(
+                        "While evaluating: %s\n\tProblemWriter_gams"
+                        " encountered an error. Ensure initial"
+                        " variable values\n\tdo not violate"
+                        " any domains (are you using log/log10?)"
```

Comment:
GAMS has presolver operations, part of which evaluate the constraints at their initial values. If a variable's level has not been explicitly set before the solve command, GAMS assumes its initial value is 0 or the closest thing to 0. So, if there is a constraint written into the GAMS model file with, for example, a log function, when GAMS is told to solve it will evaluate that constraint at initial conditions, and if that log function is called on something that evaluates to 0 or less, GAMS will throw an execution error. I don't know of any way around that behavior from GAMS, so it became necessary for the user to set their variable values such that they don't violate any domains before calling the GAMS solver.